File: backend/src/main/java/com/ecommerce/backend/service/OrderService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.request.CreateOrderRequest;
import com.ecommerce.backend.dto.response.OrderResponse;

import java.util.List;

public interface OrderService {

    OrderResponse placeOrder(String tenantName,
                             Long userId,
                             CreateOrderRequest request);
    List<OrderResponse> getOrdersByTenant(String tenantName);
    List<OrderResponse> getOrdersByUser(String tenantName,
                                        Long userId);

    OrderResponse getOrderById(String tenantName,
                               Long orderId);

    void cancelOrder(String tenantName,
                     Long orderId);

    OrderResponse updateOrderStatus(String tenantName, Long orderId, String status);

    OrderResponse checkout(String tenantName, Long userId);
}

File: frontend/src/services/orderService.js
import api from "./api";

export const placeOrder = async (tenantName, userId, order) => {
  return await api.post(
    `/orders/${tenantName}/user/${userId}`,
    order
  );
};

export const checkout = async (tenantName, userId) => {
  return await api.post(
    `/orders/${tenantName}/user/${userId}/checkout`
  );
};

export const getOrders = async (tenantName, userId) => {
  return await api.get(
    `/orders/${tenantName}/user/${userId}`
  );
};

export const getTenantOrders = async (tenantName) => {
  return await api.get(
    `/orders/${tenantName}/tenant`
  );
};

export const getOrderById = async (tenantName, orderId) => {
  return await api.get(
    `/orders/${tenantName}/${orderId}`
  );
};

export const cancelOrder = async (tenantName, orderId) => {
  return await api.delete(
    `/orders/${tenantName}/${orderId}`
  );
};

export const updateOrderStatus = async (tenantName, orderId, status) => {
  return await api.patch(
    `/orders/${tenantName}/${orderId}/status`,
    null,
    { params: { status } }
  );
};